pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -781,12 +781,22 @@ func (l *LibvirtDomainManager) SyncVMI(vmi *v1.VirtualMachineInstance, useEmulat
 		if err != nil {
 			return nil, err
 		}
-		err = dom.Create()
-		if err != nil {
-			logger.Reason(err).Error("Starting the VirtualMachineInstance failed.")
-			return nil, err
+		if vmi.ShouldStartPaused() {
```

Comment:
What about something like this here?

```go
func vmiDomainCreateFlags(vmi *v1.VirtualMachineInstance) libvirt.DomainCreateFlags {
	flags := libvirt.DOMAIN_NONE

	if vmi.ShouldStartPaused() {
		flags |= libvirt.DOMAIN_START_PAUSED
	}
	return flags
}
```

...

```diff
@@ -781,21 +790,17 @@ func (l *LibvirtDomainManager) SyncVMI(vmi *v1.VirtualMachineInstance, useEmulat
+               createFlags := vmiDomainCreateFlags(vmi)
+               err = dom.CreateWithFlags(createFlags)
+               if err != nil {
+                       logger.Reason(err).
+                               Errorf("Starting the VirtualMachineInstance with flags %v.", createFlags)
+                       return nil, err
+               }
+
                if vmi.ShouldStartPaused() {
-                       err = dom.CreateWithFlags(libvirt.DOMAIN_START_PAUSED)
-                       if err != nil {
-                               logger.Reason(err).Error("Starting the VirtualMachineInstance in paused state failed.")
-                               return nil, err
-                       }
                        l.paused.add(vmi.UID)
                        logger.Info("Domain started in paused state.")
-               } else {
-                       err = dom.Create()
-                       if err != nil {
-                               logger.Reason(err).Error("Starting the VirtualMachineInstance failed.")
-                               return nil, err
-                       }
-                       logger.Info("Domain started.")
                }
```

So that you move the logic (and any future one in case there will be more values for the startStrategy field) inside the `vmiDomainCreateFlags()` and you always create the domain with the same (more-specific) `dom.CreateWithFlags()` call with a bit less clutter?

Reject or change names freely :)

Reply:
Thank you. Changed.